src/frontend/utils.ts

```diff
@@ -21,59 +21,71 @@ export default class FrontendUtilities extends Utilities {
   }
 
   /**
-  * Sends a request to real BTT built in webserver with given data translated as GET query params
-  */
+   * Sends a request to real BTT built in webserver with given data translated as GET query params
+   */
   public async callBetterTouchTool(
-    action: string, 
+    action: string,
     data: Types.BttPayload,
     config: Types.AppConfig,
-    translate: boolean = true,
+    translate: boolean = true
   ): Promise<Types.CallResult> {
     if (this.isInBttWebView) {
+      if (config.sharedSecret) {
+        data["shared_secret"] = config.sharedSecret;
+      }
       return this.callBttWebViewFunctions(action, data, translate);
     }
     return this.callWebserverApi(action, data, config, translate);
   }
 
-  public performanceNow = () => { return window.performance.now(); };
+  public performanceNow = () => {
+    return window.performance.now();
+  };
 
   private get isInBttWebView(): boolean {
     const WVWindow: Types.WebViewWindow = window as Types.WebViewWindow;
-    return (Boolean(WVWindow.BTT) && typeof WVWindow.BTT.callHandler === 'function');
+    return (
+      Boolean(WVWindow.BTT) && typeof WVWindow.BTT.callHandler === "function"
+    );
   }
 
   /**
    * Available only on frontend, because this is valid only for built in BTT webview
-   * @param action 
-   * @param data 
-   * @param translate 
+   * @param action
+   * @param data
+   * @param translate
    */
   private callBttWebViewFunctions(
     action: string,
     data: Types.BttPayload,
-    translate: boolean,
+    translate: boolean
   ): Promise<Types.CallResult> {
     const start = Number(this.performanceNow().toFixed(3)) * 100;
     const WVWindow: Types.WebViewWindow = window as Types.WebViewWindow;
     let payload: Types.BttPayload = data;
 
     if (translate) {
-      const properBttNotation = this.translateObjectKeysToBttNotation(data.json);
-      payload = { json: JSON.stringify(properBttNotation) }
-    };
+      const properBttNotation = this.translateObjectKeysToBttNotation(
+        data.json
+      );
+      payload = { ...payload, json: JSON.stringify(properBttNotation) };
+    }
 
     return new Promise((res, rej) => {
-      let timeout = setTimeout(() => rej('Error: timedout at webview window function'), 30000);
- 
+      let timeout = setTimeout(
+        () => rej("Error: timedout at webview window function"),
+        30000
+      );
+
       WVWindow.BTT.callHandler(action, payload, (result: any) => {
         const end = Number(this.performanceNow().toFixed(3)) * 100;
         clearTimeout(timeout);
         res({
-          time: (end / 100) - (start / 100),
+          time: end / 100 - start / 100,
```

Comment:
I liked parenthesis around here, I know that probably they were autoremoved, but can you please restore them? Surely we'd have to set up proper rule for that as well

Reply:
i will create 2 new pull requests. 1 for shared secret and another one for the tslint + prettier config ;)